content.md

```diff
@@ -1,23 +1,278 @@
+
+```bash
+gem install rails # will download the latest version of rails
+```
+
+Once the install is complete close your terminal and launch a new one for changes to take effect
+
+And now we can verify Rails is installed:
+
+```bash
+rails -v
+# Rails 7.1.3
+```
+
+## Install PostgreSQL
+
+You can install [PostgreSQL](http://www.postgresql.org/) server and client from Homebrew:
+
+```bash
+brew install postgresql
 ```
-![](assets/example-image.png)
+
+Once this command is finished, it gives you a couple commands to run. Follow the instructions and run them:
+
+```bash
+# To have launchd start postgresql at login:
+brew services start postgresql
 ```
 
-You can use the path `/assets/my-image.png` or `assets/my-image.png`, both will render in your local markdown preview; and when you connect the repository with a Learn Lesson, the assets will upload to Cloudinary and the paths will automatically be converted to a hosted URL, e.g.:
+By default the postgresql user is your current macOS username with no password. For example, my macOS user is named `pttl-42` so I can login to postgresql with that username.
+
+```bash
+psql pttl-42
+```
+
+## GUI Install Alternatives
+
+[Postgres.app](https://postgresapp.com/)
+
+[Postico](https://eggerapps.at/postico/v1.php)
 
+## Setup Version Control
+
+We'll be using Git for our version control system so we're going to set it up to match our [Github](https://github.com/) account. If you don't already have a Github account, make sure to [register](https://github.com/). It will come in handy for the future.
+
+Replace the example name and email address in the following steps with the ones you used for your Github account.
+
+```bash
+git config --global color.ui true
+git config --global user.name "YOUR NAME" # swap "YOUR NAME" for your Github username
+git config --global user.email "YOUR@EMAIL.com" #swap "YOUR@EMAIL.com" for the email associated with your Github account
+ssh-keygen -t ed25519 -C "YOUR@EMAIL.com"
 ```
-![](https://res.cloudinary.com/[CLOUD_NAME]/image/upload/[IMAGE_VERSION]/appdev-lessons/[REPO_NAME]/[BRANCH]/[IMAGE_NAME])
+
+The last command will create a ssh key that is unique to your machine. The next step is to take the newly generated SSH key and add it to your Github account. You want to copy and paste the output of the following command and [paste it here](https://github.com/settings/ssh).
+
+```bash
+cat ~/.ssh/id_ed25519.pub
 ```
+
+Once you've done this, you can check and see if it worked:
+
+```bash
+ssh -T git@github.com
+```
+
+You should get a message like this:
+
+```bash
+Hi {github username}! You've successfully authenticated, but GitHub does not provide shell access.
+```
+
+## Conclusion
+
+In this lesson, we've covered the essential steps to set up a local developer environment on macOS:
+
+1. **IDE Installation**: We installed Visual Studio Code (VSCode) and configured the `code` command in the PATH to open VSCode from the terminal conveniently.
```

Comment:
I think this list can be 1 sentence. Maybe add a bit more info on how it might be different depending on your technical stack and project requirements. So consult the project README.

Reply:
Added sentence explaining this IDE is tailored towards use in the SDF course.